### DISCUSSION TITLE
Scalar UI plugins support

### DISCUSSION THREAD

**Discussion by jvmlet** [scalar/scalar#4246] [Ideas]
While  looking for alternatives to Swashbuckle, I'm after the  UI plugin support, that, for example, can intercept  http requests and add UI components.
This is what we have for SwaggerUI, as an example of plugin capabilities:
![image](https://github.com/user-attachments/assets/996eeb64-5f25-42b1-9ad5-d6566a4d98ca)
 `Manual` option enables the default `Authorize` button and all the functionality it provides.
 `Auto`  disables default `Authorize` button ,  adds the http request interceptor for API call invocation, simelesly  generates auth token using kerberos (another rest call, configured from server side as UI plugin) and automatically propagates  the generated token as  `Authorization` header  to original  request.

Can this  **client-side** functionality  be achieved with Scalar ?

**Comment by hanspagel** [comment]
Cool idea!

**Reply by jvmlet** [reply]
thanks, but not mine ;-) `SwaggerUI`  has it for years
